### ISSUE TITLE
Revise "Playground" story approach in the `@carbon/react` storybook

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#16816]
## Context

> As a developer using `@carbon/react`,
> I need a way to copy and paste valid/working component code snippets from storybook,
> So that I can get a jumpstart using a component in my project, experiment in a sandbox/stackblitz, etc.
> 

The `@carbon/react` storybook has supported this use case so far via the [storysource plugin](https://storybook.js.org/addons/@storybook/addon-storysource/). It displays the code source of the current story being shown in the canvas. It meets this need fairly well if the story uses explicit values within the story.

<details><summary>Click to expand example</summary>

![CleanShot 2024-06-18 at 13 42 30@2x](https://github.com/carbon-design-system/carbon/assets/3360588/ebd0cc25-cf79-4635-af34-3576da3b473d)

</details> 

Although when controls/args are used, the source code doesn't work well for copy/paste because it contains storybook-specific code (`...args`) that would cause an error in any environment outside of our storybook:

<details><summary>Click to expand example</summary>

![CleanShot 2024-06-18 at 13 38 46@2x](https://github.com/carbon-design-system/carbon/assets/3360588/be5d7524-6914-42c6-9afb-473c9c4ce6ab)

</details> 

## As-is experience

To combat this we have so far tried to follow a strategy where we only use controls/args on a single specific story named "Playground". Each component typically has one - controls are fully enabled, and other stories have explicit prop usage instead of controls. This way the story source is copy/paste-able for every story except the Playground story.

In some cases Playground stories can't be used because the component has variants that require different stories and aren't able to be contained in one Playground story. Button and StructuredList are an example of this currently. In these cases controls are enabled on every story and there is no Playground story.

This strategy and it's inconsistency is confusing for both maintainers and users of the storybook.

## Proposal

Stories are [intended to be authored using args](https://storybook.js.org/docs/writing-stories#how-to-write-stories). Every story should be updated to use args where possible. We could then delete the playground stories. 

Another way to think about this is that we're renaming `Playground` stories to be `Default`. The goal isn't to encourage adding a new story for every variant of a component - instead this change encourages further reduction of stories towards greater reliance on args. 

There are a few ways we can continue supporting the copy/paste use case:
* Enhance our docs pages to include [Source](https://storybook.js.org/docs/api/doc-block-source) or [Canvas](https://storybook.js.org/docs/api/doc-block-canvas) blocks, which appear to support dynamic code snippet generation when using args via the [`type` option](https://storybook.js.org/docs/api/doc-block-source#type):
  * > dynamic: Renders the story source with dynamically updated arg values
* Provide an "Open in Stackblitz" link on every story via [the stackblitz addon](https://storybook.js.org/addons/@stackblitz/storybook-addon-stackblitz), where the current story source can be opened in a sandbox for experimentation

Additional benefits of this approach include:
* Having one less story in every component folder
* A simplified story authoring experience that would match the storybook documentation
* Every story would support args, so we could use args for configuration in AVT and VRT tests, which would simplify the "arrange" portion of these tests

**Comment by sstrubberg** [comment]
I freaking love quality of life improvements like this. Let's figure out a way to slide this in. Might be a fun EOY project.

**Comment by tay1orjones** [comment]
Agree! It's kind of related to https://github.com/carbon-design-system/carbon/issues/12947, probably something to do alongside it actually.

**Comment by tay1orjones** [comment]
There's still a ton of playground stories in the storybook right now. I believe we're aiming for this to be part of the work in https://github.com/carbon-design-system/carbon/issues/18344 but I'm not seeing it as a part of the acceptance criteria there.

**Comment by guidari** [comment]
We still have 4 playground stories on storybook. However those were left intentionally, because to me they make sense, since the default stories in some cases couldn't fit controls. We can talk about it to decide what to do with those left, maybe we can change the names or something...

The Status Indicator is a new component added after this workstream.

![Image](https://github.com/user-attachments/assets/2326eae5-4cd7-4d17-94e6-fcba743062a5)
